syntax = "proto3";

package ai_inference;

// 서비스 정의
service AIInference {
  rpc DetectObjects (ImageRequest) returns (ObjectDetectionResponse);
  rpc RecognizeFaces (ImageRequest) returns (FaceRecognitionResponse);
}

// 공통 이미지 데이터 요청 메시지
message ImageRequest {
  string image_id = 1;
}

// 객체 인식 결과 메시지
message ObjectDetectionResponse {
  string object_name = 1;
  float confidence = 2;
  BoundingBox box = 3;
}

// 바운딩 박스 메시지
message BoundingBox {
  int32 x = 1;
  int32 y = 2;
  int32 width = 3;
  int32 height = 4;
}

// 얼굴 인식 결과 메시지
message FaceRecognitionResponse {
  string person_type = 1; // 예: "Employee", "Guest", "Unknown"
  optional string employee_id = 2; // 직원의 경우 ID, Guest 또는 Unknown의 경우 비어 있을 수 있음
  float confidence = 3;
}
